conbench/api/history.py

```diff
@@ -37,8 +26,19 @@ def get(self, benchmark_id):
         tags:
           - History
         """
-        history = self._get(benchmark_id)
-        return self.serializer.many.dump(history)
+        # TODO: think about the case where samples if of zero length. Can this
+        # happen? If it can happen: which response would we want to emit to the
+        # HTTP client? An empty array, or something more convenient?
```

Comment:
@austin3dickey when you think about specifying the history api endpoint: zero length result allowed or not?

Reply:
Wouldn't we at least want to always return one sample corresponding to the given `benchmark_id`?

Reply:
> Wouldn't we at least want to always return one sample corresponding to the given benchmark_id?

Ah :) I didn't even think that "far" lol. You are saying: if benchmark_id is known that we can emit at least one sample; in all other cases we can have decent error cases (benchmark ID not known, benchmark known but doesn't contain samples (err state?), ...); in those cases we return an error response (and do not need to show an empty array).

Reply:
Yes, exactly! :)